src/binder/textualidentityparser.cpp

```diff
@@ -417,12 +439,64 @@ namespace BINDER_SPACE
                 cursor++;
             }
 
-            if (dwFoundNumbers == static_cast<DWORD>(iRequiredVersionParts))
+            // Compat with .NET Framework:
+            // - If no value is provided, the version is considered invalid.
+            //   Example: "MyAssembly, Version="
+            // - If the major version is provided and the remaining components are valid, the version is considered valid.
+            //   Examples:
+            //     "MyAssembly, Version=1"
+            //     "MyAssembly, Version=1."
+            //     "MyAssembly, Version=1.."
+            //     "MyAssembly, Version=1..."
+            if (dwFoundNumbers == 0)
             {
-                pAssemblyVersion->SetFeatureVersion(dwNumbers[0], dwNumbers[1]);
-                pAssemblyVersion->SetServiceVersion(dwNumbers[2], dwNumbers[3]);
-                fIsValid = TRUE;
+                goto Exit;
             }
+
+            if (dwFoundNumbers < 2 || dwNumbers[0] == UnspecifiedNumber || dwNumbers[1] == UnspecifiedNumber)
+            {
+                // Compat with .NET Framework: If the major or minor version is unspecified, all components are considered
+                // unspecified. Set the major version to the unspecified value (this is a special value indicating that the
+                // whole version is unspecified, see AssemblyName::HaveAssemblyVersion).
+                //
+                // Examples:
+                //   "MyAssembly, Version=."
+                //   "MyAssembly, Version=.1"
+                //   "MyAssembly, Version=1"
+                //   "MyAssembly, Version=1."
+                dwNumbers[0] = UnspecifiedNumber;
+                dwNumbers[1] = 0;
+                dwNumbers[2] = 0;
+                dwNumbers[3] = 0;
+            }
+            else
+            {
+                // Difference from .NET Framework:
+                // - In .NET Framework, unspecified components retain the unspecified value. This special value is treated as
+                //   unspecified when loading an assembly, even if explicitly specified as 65535 as part of the version.
+                // - In .NET Core, any unspecified version component after the minor version is given a value of zero. When
+                //   loading an assembly, .NET Core will match and load an assembly that has a version that is >= the requested
```

Comment:
It looks like new Version(1, 1) sets the other two components to (int)-1. It also uses int components and not ushort, and allows values greater than 65535 as opposed to the AssemblyName(string) constructor. BaseAssemblySpec is using ushort for the version components.

What do we want to do about that? I'm thinking one way is to make them all int and use (int)-1 to indicate an unspecified component. Or we could leave that parsing difference in place and sign-extend the ushort into int when creating the Version object.